### PR TITLE
Added emtpy validation

### PR DESCRIPTION
**Pull request by TheTechArch**
More resource improvement

### CODE REVIEW

File: src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs [hunk 1]
```diff
@@ -53,66 +53,87 @@ public static bool ValidateResource(ServiceResource serviceResource, out Diction
                 isValid = false;
             }
 
+            if (serviceResource.ResourceType.Equals(ResourceType.Default))
+            {
+                AddValidationMessage(validationMessages, "ResourceType", "ResourceType needs to be set");
+                isValid = false;
+            }
+
+            if (serviceResource.HasCompetentAuthority == null ||
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs:62]: I'm not entirely sure how to interpret this check. It fails if:

1. HasCompetentAuthority is null
2. (Orgcode is set to something other than "ttd" AND organization is unset

Did I read that correctly?

In any case, this should probably be rewritten with either named functions or just variables in front of the if:

```csharp
// just as an example
bool hasNonTtdOrgcode = string.IsNullOrEmpty(orgcode) && orgCode.Equals("ttd", StringComparison.OrdinalIgnoreCase);
```

File: src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs [hunk 2]
```diff
@@ -53,66 +53,87 @@ public static bool ValidateResource(ServiceResource serviceResource, out Diction
...
 
+            if (serviceResource.ResourceType.Equals(ResourceType.Default))
+            {
+                AddValidationMessage(validationMessages, "ResourceType", "ResourceType needs to be set");
+                isValid = false;
+            }
+
+            if (serviceResource.HasCompetentAuthority == null || 
+                ((string.IsNullOrEmpty(serviceResource.HasCompetentAuthority.Orgcode) 
+                || !serviceResource.HasCompetentAuthority.Orgcode.ToLower().Equals("ttd"))
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs:64]: Don't do `.ToLower().Equals`, do `Equals(..., StringComparison.OrdinalIgnoreCase);`

File: src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs [hunk 3]
```diff
@@ -53,66 +53,87 @@ public static bool ValidateResource(ServiceResource serviceResource, out Diction
...
+
             if (!ResourceIdentifierRegex.IsMatch(serviceResource.Identifier))
             {
-                AddValidationMessage(validationMessages, "Identifier", "Invalid id. Only a-z and 0-9 is allowed together with _ and -");
+                AddValidationMessage(validationMessages, "Identifier", "Invalid id. Only a-z and 0-9 is allowed together with _ and -.  Minimum 4 characters");
                 isValid = false;
             }
 
-            if (serviceResource.Title == null || !serviceResource.Title.ContainsKey(ResourceConstants.LANGUAGE_EN))
+            if (serviceResource.Title == null || !serviceResource.Title.ContainsKey(ResourceConstants.LANGUAGE_EN) || string.IsNullOrWhiteSpace(serviceResource.Title[ResourceConstants.LANGUAGE_EN]))
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs:77]: This should be a helper function:

```csharp
if (!HasLanguage(serviceResource.Title, ResourceConstants.LANGUAGE_EN))
```

**Review by Alxandr — APPROVED** [review]

File: src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs (hunk 3 above)
Comment (TheTechArch) [reply src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs:77]: Good suggestion. Fixed

File: src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs (hunk 2 above)
Comment (TheTechArch) [reply src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs:64]: Fixed

File: src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs (hunk 1 above)
Comment (TheTechArch) [reply src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs]: Extracted method and tried to make it clear with comment

File: src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs [hunk 4]
```diff
@@ -143,6 +161,22 @@ static bool IsInvalidAppResource(ServiceResource serviceResource)
 
                 return false;
             }
+
+            /// <summary>
+            /// Validates that orgs that is not TTD needs to have orgnumber set.
+            /// </summary>
+            static bool ValidateResourceOwner(ServiceResource serviceResource)
+            {
+                if (serviceResource.HasCompetentAuthority == null ||
```
[the remark this answers is not in the document]
Comment (TheTechArch) [reply src/Altinn.ResourceRegistry.Core/Helpers/ServiceResourceHelper.cs]: Based on the above suggestion was incorrect, I restructured the code similar to suggestion to make it more clear. :-)

**Pull request merged**